applications/nrf5340_audio/src/bluetooth/ble_audio_services.c

```diff
@@ -523,6 +546,20 @@ int ble_mcs_play_pause(struct bt_conn *conn)
 	return 0;
 }
 
+int ble_mcp_conn_disconnected(struct bt_conn *conn)
+{
+	if (!IS_ENABLED(CONFIG_BT_MCC)) {
+		LOG_ERR("MCC not enabled");
+		return -ECANCELED;
+	}
+
+	uint8_t idx = bt_conn_index(conn);
+
+	LOG_DBG("MCS discover state reset due to disconnection");
+	mcp_mcs_disc_status[idx] = IDLE;
+	return 0;
+}
```

Comment:
You could guard this entire function like
```suggestion
#if (CONFIG_BT_MCC)
int ble_mcp_conn_disconnected(struct bt_conn *conn)
{
	uint8_t idx = bt_conn_index(conn);

	LOG_DBG("MCS discover state reset due to disconnection");
	mcp_mcs_disc_status[idx] = IDLE;
	return 0;
}
#endif /* CONFIG_BT_MCC */
```

and then where you call it, you can guard the call with `IS_ENABLED(CONFIG_BT_MCC)`. Not sure if that is how it is usually done in NCS, but at least that's the way it is done in Zephyr

Reply:
Good point Emil. I will create a task for a general cleanup of this.